### PR TITLE
Added additional MIM validations and warning for modulestest

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Solidified MIM parsing for modulestest, will have warnings when parsing invalid MIMs
* Installed osconfig on modulestest CI VM hosts to enable `Configuration` module tests

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/test/main.cpp [hunk 1]
```diff
@@ -89,10 +89,15 @@ TestRecipes LoadValuesFromConfiguration(std::stringstream& ss, std::string modul
 
             recipeMetadata.m_recipeModuleSessionLoader->Load(modulePaths);
             TestRecipes recipes = TestRecipeParser::ParseTestRecipe(recipeMetadata.m_testRecipesPath);
+            pMimObjects mimObjects = MimParser::ParseMim(recipeMetadata.m_mimPath);
+            if (nullptr == mimObjects)
+            {
+                TestLogError("WARNING, could not parse mim file '%s'. MIM validation will be skipped for '%s'", recipeMetadata.m_mimPath.c_str(), recipeMetadata.m_modulePath.c_str());
```
Comment (MariusNi) [inline src/modules/test/main.cpp:95]: >mim [](http://example.com/codeflow?start=55&length=3)

MIM (uppercase). All lowercase means something else.

File: src/modules/test/main.cpp [hunk 2]
```diff
@@ -117,10 +122,15 @@ TestRecipes LoadValuesFromCLI(char* modulePath, char* mimPath, char* testRecipes
 
     recipeMetadata.m_recipeModuleSessionLoader->Load(std::vector<std::string>{modulePath});
     TestRecipes recipes = TestRecipeParser::ParseTestRecipe(recipeMetadata.m_testRecipesPath);
+    pMimObjects mimObjects = MimParser::ParseMim(recipeMetadata.m_mimPath);
+    if (nullptr == mimObjects)
+    {
+        TestLogError("WARNING, could not parse mim file '%s'. MIM validation will be skipped for '%s'", recipeMetadata.m_mimPath.c_str(), modulePath);
```
Comment (MariusNi) [inline src/modules/test/main.cpp:128]: >mim [](http://example.com/codeflow?start=47&length=3)

Same as above (MIM, uppercase letters) pls.

File: src/modules/test/RecipeInvoker.cpp [hunk 3]
```diff
@@ -83,8 +83,10 @@ void RecipeInvoker::TestBody()
...
-                EXPECT_TRUE(json_value_equals(recipe_payload, returned_payload)) << "Non matching recipe payload" << std::endl << "Recipe   payload: " << m_recipe.m_payload << std::endl << "Returned payload: " << recipe_payload_str << std::endl;
+                EXPECT_NE(nullptr, recipe_payload) << "Failed to parse recipe payload" << std::endl << "JSON: " << m_recipe.m_payload.c_str() << std::endl;
+                EXPECT_NE(nullptr, returned_payload) << "Failed to parse returned payload" << std::endl << "JSON: " << recipe_payload_str.c_str() << std::endl;
+                EXPECT_EQ(json_value_get_type(returned_payload), json_value_get_type(recipe_payload)) << "Non matching payload types. Recipe payload: " << json_value_get_type(recipe_payload) << ", returned payload: " << json_value_get_type(returned_payload) << std::endl;
+                EXPECT_EQ(json_value_equals(recipe_payload, returned_payload), 1) << "Non matching recipe payload" << std::endl << "Recipe   payload: " << m_recipe.m_payload << std::endl << "Returned payload: " << recipe_payload_str.c_str() << std::endl;
```
Comment (MariusNi) [inline src/modules/test/RecipeInvoker.cpp:89]: >Recipe   payload [](http://example.com/codeflow?start=132&length=16)

nit: extra space between "Recipe" and "payload"

**Review by MariusNi — APPROVED** [review]
:shipit:

File: src/modules/test/RecipeInvoker.cpp (hunk 3 above)
Comment (AhmedBM) [reply src/modules/test/RecipeInvoker.cpp:89]: that is to align it with the line below, it makes for easier JSON comparison

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**